### PR TITLE
refactor(serde_v8): move to `thiserror`, better error output

### PR DESCRIPTION
**Pull request by attila-lin**
refs to #17318

### CODE REVIEW

File: serde_v8/error.rs [hunk 1]
```diff
@@ -1,54 +1,54 @@
...
+  #[error("serde_v8 error: invalid type, expected: buffer")]
   ExpectedBuffer,
+  #[error("serde_v8 error: invalid type, expected: detachable")]
   ExpectedDetachable,
+  #[error("serde_v8 error: invalid type, expected: external")]
   ExpectedExternal,
 
+  #[error("serde_v8 error: invalid type, expected: utf8")]
   ExpectedUtf8,
+  #[error("serde_v8 error: invalid type, expected: latin1")]
```
Comment (bartlomieju) [inline serde_v8/error.rs:37]: Can you make these errors show what was actual type?

Comment (attila-lin) [reply serde_v8/error.rs:37]: Let me try~

**Review by aapoalas — DISMISSED** [review]
Can you rebase the changes?

File: serde_v8/error.rs (hunk 1 above)
Comment (bartlomieju) [reply serde_v8/error.rs:37]: This will be handled by @levex as part of https://github.com/denoland/deno/issues/17406

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @attila-lin this is very helpful!

**Pull request merged**